### PR TITLE
API: add toggle keep build forever

### PR DESCRIPTION
**Pull request by trevormccasland**
This commit aims to add an API function that allows a user of the
API to toggle the "keep this build forever" flag on the jenkins
build page.

This is related to #645 but does not close it as it's missing the
"edit the build information - display name and description"
part of the summary.

https://github.com/pycontribs/jenkinsapi/issues/645

### CODE REVIEW

**Comment by trevormccasland** [comment]
api.py does not have a unit testing file from what I found, do we want to add one or should I add a test else where? I didn't see api.py being imported in this project. https://github.com/pycontribs/jenkinsapi/search?utf8=%E2%9C%93&q=%22import+api%22&type=

**Comment by lechat** [comment]
I am not sure why `api.py` is there. Could you please add a method to `Build` class instead?

File: jenkinsapi/build.py [hunk 1]
```diff
@@ -531,3 +532,16 @@ def get_env_vars(self):
                           'is installed.')
             raise ex
         return data['envMap']
+
+    def toggle_keep_build_forever(self):
+        """
+        Toggles the keep this build forever flag for the specified build.
+        """
+        requester = self.job.jenkins.requester
+        if not isinstance(requester, CrumbRequester):
```
Comment (lechat) [inline jenkinsapi/build.py]: Please leave `requester` as is - not all people use it and if you set it here, this will break on Jenkins where CSRF protection is switched off.

File: jenkinsapi_tests/unittests/test_build.py [hunk 2]
```diff
@@ -215,3 +216,11 @@ def fake_get_data(cls, tree=None, params=None):
             build.get_env_vars()
     assert '' == str(excinfo.value)
     assert len(record) == 0
+
+
+def test_toggle_keep_build_forever(monkeypatch, build):
```
Comment (lechat) [inline jenkinsapi_tests/unittests/test_build.py]: Please put that test into systests - this is to ensure that it will work on actual Jenkins.

File: jenkinsapi/api.py [hunk 3]
```diff
@@ -244,3 +245,14 @@ def search_artifact_by_regexp(jenkinsurl, jobid, artifactRegExp,
...
 
     raise ArtifactsMissing()
+
+
+def toggle_keep_build_forever(jenkinsurl, jobname, build_no,
+                              username=None, password=None, ssl_verify=True):
+    """
+    Toggles the keep this build forever flag for the specified build
+    """
+    crumb = CrumbRequester(username, password, baseurl=jenkinsurl,
```
Comment (lechat) [inline jenkinsapi/api.py]: Sorry, but not going to accept this change. This will not work in Jenkins where CSRF protection is switched off. And this code needs to go into `Build` object.

File: jenkinsapi/api.py [hunk 4]
```diff
@@ -248,9 +248,9 @@ def search_artifact_by_regexp(jenkinsurl, jobid, artifactRegExp,
 
 
 def toggle_keep_build_forever(jenkinsurl, jobname, build_no,
-                              username=None, password=None, ssl_verify=True):
+                              username, password, ssl_verify=True):
```
Comment (lechat) [inline jenkinsapi/api.py]: Why this change is required?

**Comment by lechat** [comment]
I built it myself in #672
